### PR TITLE
Implement cross-platform URL opening convenience func

### LINKED ISSUE #63: Add cross-platform compatible URL opening function
We find ourselves wanting to open a URL on behalf of a user via our Golang tooling. 

This open routine must be cross-platform so that it works seamlessly for all our users regardless of their chosen platform.

### PR DESCRIPTION
**Pull request by zackproser**
## Description

Closes #63 

## TODOs

Please ensure all of these TODOs are completed before asking for a review.

- [x] Ensure the branch is named correctly with the issue number. e.g: `feature/new-vpc-endpoints-955` or `bug/missing-count-param-434`.
- [ ] Update the docs.
- [x] Keep the changes backward compatible where possible.
- [x] Run the pre-commit checks successfully.
- [x] Run the relevant tests successfully.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.

### CODE REVIEW

File: url/url.go [hunk 1]
```diff
@@ -52,3 +54,24 @@ func mergeQuery(originalQuery url.Values, newQuery url.Values) url.Values {
 func stripSlashes(str string) string {
 	return strings.Trim(str, "/")
 }
+
+// Attempt to open a URL in the user's browser. We use this to open docs, PRs we've
+// programmatically opened, etc
+func openURL(url string) error {
```
Comment (yorinasub17) [inline url/url.go:60]: Make this public (`OpenURL`)

**Review by yorinasub17 — COMMENTED** [review]
LGTM! Just need to make it a public function so it is callable.

File: url/url.go (hunk 1 above)
Comment (zackproser) [reply url/url.go:60]: Ah yeah that would probably help! :face_exhaling:

Comment (zackproser) [reply url/url.go]: Done in ba66eab

**Comment by zackproser** [comment]
Thanks for review! Going to merge this in now.

**Pull request merged**